### PR TITLE
test: Make `util/test_runner.py` honor `BITCOINUTIL` and `BITCOINTX`

### PR DESCRIPTION
**Pull request by hebasto**
This PR is a continuation of changes to our testing frameworks (https://github.com/bitcoin/bitcoin/pull/27554, https://github.com/bitcoin/bitcoin/pull/27561) that allow them to work correctly in a multi-config build environment that is possible for [upcoming](https://github.com/bitcoin/bitcoin/pull/25797) CMake-based build system. That means that built for different configurations binaries (e.g., "Debug" and "Release") can coexist in separated directories.

The commit has been pulled from https://github.com/hebasto/bitcoin/pull/15 and it seems [useful](https://github.com/hebasto/bitcoin/pull/15#discussion_r1200251404) by itself as:
> I believe the rationale for allowing to drop in the executables via env var is to allow to test the guix-produced, or other third-party-produced executables...

The current implementation of the `test/functional/test_framework/test_framework.py` script uses the same approach: https://github.com/bitcoin/bitcoin/blob/09351f51d279612973ecd76811dc075dff08209f/test/functional/test_framework/test_framework.py#L231-L246

### CODE REVIEW

File: test/util/test_runner.py [hunk 1]
```diff
@@ -74,6 +74,11 @@ def bctest(testDir, testObj, buildenv):
     """
     # Get the exec names and arguments
     execprog = os.path.join(buildenv["BUILDDIR"], "src", testObj["exec"] + buildenv["EXEEXT"])
+    if testObj["exec"] == "./bitcoin-util":
```
Comment (theuni) [inline test/util/test_runner.py:77]: Shouldn't this (and similar for the other) append `EXEEXT` to `./bitcoin-util` here?

**Review by theuni — COMMENTED** [review]
Concept ACK.

I initially complained about the approach [here](https://github.com/hebasto/bitcoin/pull/15#discussion_r1192682955), but it seems it's just following the current convention.

File: test/util/test_runner.py (hunk 1 above)
Comment (hebasto) [reply test/util/test_runner.py:77]: I guess not. This line is about parsing of the [`test/util/data/bitcoin-util-test.json`](https://github.com/bitcoin/bitcoin/blob/master/test/util/data/bitcoin-util-test.json) file.

The correct file extension has already been added in the `execprog` variable above, or provided in an environment variable.

**Comment by hebasto** [comment]
As some progress is being made in the review of the coupled https://github.com/hebasto/bitcoin/pull/15, friendly ping @MarcoFalke @stickies-v @ryanofsky :)

**Review by sedited — APPROVED** [review]
ACK 4f2f615d1362afe92cabe9eab50087f8bfe454fd

The error reporting in this file is a bit opaque imo, but that is orthogonal to the work here.

**Review by stickies-v — APPROVED** [review]
ACK 4f2f615d1362afe92cabe9eab50087f8bfe454fd

**Pull request merged**